internal/controller/runtime/fsm/runtime_fsm_patch_shoot.go

```diff
@@ -76,6 +76,11 @@ func sFnPatchExistingShoot(ctx context.Context, m *fsm, s *systemState) (stateFn
 				return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
 			}
 
+			if k8serrors.IsForbidden(err) {
```

Comment:
In general the solution is OK, but I would add some comment here. Usually `forbidden` is an unretryable error so it may be confusing at the first glance.

Reply:
Good point. I will explain that.